src/platform/input/game-pads.js

```diff
@@ -2,6 +2,17 @@ import { EventHandler } from '../../core/event-handler.js';
 import { EVENT_GAMEPADCONNECTED, EVENT_GAMEPADDISCONNECTED, PAD_FACE_1, PAD_FACE_2, PAD_FACE_3, PAD_FACE_4, PAD_L_SHOULDER_1, PAD_R_SHOULDER_1, PAD_L_SHOULDER_2, PAD_R_SHOULDER_2, PAD_SELECT, PAD_START, PAD_L_STICK_BUTTON, PAD_R_STICK_BUTTON, PAD_UP, PAD_DOWN, PAD_LEFT, PAD_RIGHT, PAD_VENDOR, XRPAD_TRIGGER, XRPAD_SQUEEZE, XRPAD_TOUCHPAD_BUTTON, XRPAD_STICK_BUTTON, XRPAD_A, XRPAD_B, PAD_L_STICK_X, PAD_L_STICK_Y, PAD_R_STICK_X, PAD_R_STICK_Y, XRPAD_TOUCHPAD_X, XRPAD_TOUCHPAD_Y, XRPAD_STICK_X, XRPAD_STICK_Y } from './constants.js';
 import { math } from '../../core/math/math.js';
 
+/**
+ * Fetch Gamepads from API.
+ *
+ * @type {Function}
+ * @returns {Gamepad[]} Retrieved gamepads from the device.
+ * @ignore
+ */
+const fetchGamepads = (navigator.getGamepads || navigator.webkitGetGamepads || function () {
```

Comment:
I worked this out in my PR which makes sure that 1) no garbage is produced by reusing `[]` while 2) supporting node, this looks like it's accessing navigator without

Reply:
Yeah but as your PR is still not live so I cannot use it... Once it gets merged I'll be more than happy to use it!
Regarding 2 I already corrected it right after this commit, you are probably reviewing on an older version...

Reply:
Yea, I just realized, for some reason the entire file in "Files changed" was still in the old state, so I could comment on that line